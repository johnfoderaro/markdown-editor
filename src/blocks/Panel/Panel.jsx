import React from 'react';
import { hot } from 'react-hot-loader';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Block = styled.div`
  padding-right: 1rem;
  padding-left: 1rem;
  height: 100%;
  width: ${({ type }) => (type === 'explorer' ? '20%' : '80%')};
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  background: ${({ type }) => (type === 'explorer' ? '#eee' : '#fff')};
`;

const Panel = ({ children, type }) => (
  <Block type={type}>
    {children}
  </Block>
);

Panel.propTypes = {
  type: PropTypes.string.isRequired,
  children: PropTypes.arrayOf(PropTypes.element).isRequired,
};

export default hot(module)(Panel);
