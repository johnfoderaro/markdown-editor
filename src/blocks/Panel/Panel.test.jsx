import React from 'react';
import { shallow } from 'enzyme';

import Panel from './Panel';

let tree;

beforeAll(() => {
  tree = shallow(
    <Panel>
      <h1>one</h1>
      <p>two</p>
    </Panel>,
  );
});

describe('<Panel />', () => {
  it('should be defined', () => {
    expect(tree).toBeDefined();
  });
  it('should render correctly', () => {
    expect(tree).toMatchSnapshot();
  });
  it('should render children correctly', () => {
    expect(tree.find('h1').text()).toEqual('one');
    expect(tree.find('p').text()).toEqual('two');
  });
});
