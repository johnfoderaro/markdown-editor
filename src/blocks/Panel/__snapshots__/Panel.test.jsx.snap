// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Panel /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Panel>
    <h1>
      one
    </h1>
    <p>
      two
    </p>
  </Panel>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h1>
          one
        </h1>,
        <p>
          two
        </p>,
      ],
      "type": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "one",
        },
        "ref": null,
        "rendered": "one",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "two",
        },
        "ref": null,
        "rendered": "two",
        "type": "p",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  padding-right: 1rem;
  padding-left: 1rem;
  height: 100%;
  width: ",
          [Function],
          ";
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  background: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h1>
            one
          </h1>,
          <p>
            two
          </p>,
        ],
        "type": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "one",
          },
          "ref": null,
          "rendered": "one",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "two",
          },
          "ref": null,
          "rendered": "two",
          "type": "p",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
  padding-right: 1rem;
  padding-left: 1rem;
  height: 100%;
  width: ",
            [Function],
            ";
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  background: ",
            [Function],
            ";
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
